### DISCUSSION TITLE
Why I get null when reselecting radio input option?

### DISCUSSION THREAD

**Discussion by angelhodar** [react-hook-form/react-hook-form#9488] [Q&A]
I have a problem that I cant replicate in a codesandbox and I would like to know if someone has run into a similar problem or has any idea of what could be wrong.

Here is the codesandbox: https://codesandbox.io/s/confident-bartik-zhvy2h?file=/src/App.js

It just consists in a Form and an Input component that is a radio input. In my current app i set the initial state for `months` to be 6, so if I change it to 12 the form state is changed properly, BUT if then i change it back to 6 I get a `null` value. As i said i cant replicate it in the codesandbox so i have no idea what can be wrong. The options are set properly and the initial state is also correct, thats why I am so confused. Here you have a gif of the problem:

![bug](https://user-images.githubusercontent.com/26228709/205080386-70c6d318-578b-457b-bb2c-f8f75bd540e0.gif)

**Comment by bluebill1049** [comment]
Not sure, it works for me in your demo: 

<img width="772" alt="Screenshot 2022-12-02 at 8 51 23 am" src="https://user-images.githubusercontent.com/10513364/205166541-ddfdb3bb-abae-405f-b4f1-359f83548bba.png">

**Reply by angelhodar** [reply]
Yes thats why I am so lost to find the problem. Do you have any idea of what could be the cause? Maybe I am not setting the values of the input radios properly?

**Reply by bluebill1049** [reply]
not on top of my head, i couldn't reproduce the problem in your codesndbox as well, but you can try to debug on your `onChange` and see what value is sending back to hook form.

**Comment by fknop** [comment]
I'm facing the same issue in my app when resetting the form with new values. Although the value is correct, the useWatch even returns the proper value, but when calling `getValues()` the value is null

**Reply by fknop** [reply]
I found out what my issue was.. A component was setting the `checked` attribute manually on the input :)
Nothing to see here
